### PR TITLE
Incorrect generation for xmlObject when arrays are serialized with XM…

### PR DESCRIPTION
**Pull request by emilw**
…L serializer.

Closes https://github.com/RSuter/NJsonSchema/issues/777

### CODE REVIEW

**Comment by emilw** [comment]
@Rsuter, do you have any feedback on this? I have been testing it out end to end with NSwag also to see the full result besides the updated Unit tests

File: src/NJsonSchema/Infrastructure/XmlObjectExtension.cs [hunk 1]
```diff
@@ -29,14 +29,35 @@ public static void GenerateXmlObjectForType(this JsonSchema4 schema, Type type)
...
         }
+
+        /// <summary>
+        /// Generate XML object for a JSON Schema definition
+        /// </summary>
+        /// <param name="schema">The JSON Schema</param>
+        /// <param name="type">The array type</param>
+        public static void GenerateXmlObjectForArrayType(this JsonSchema4 schema, Type type)
+        {
+            if (schema.IsArray && schema.ParentSchema == null)
```
Comment (RSuter) [inline src/NJsonSchema/Infrastructure/XmlObjectExtension.cs]: Why `schema.ParentSchema == null`? I think it is not completely safe to rely on ParentSchema...

**Comment by RSuter** [comment]
Thanks for the PR. Looks good, but to be honest I dont really know the XML feature... did you ensure that you just fixed array handling and didnt break existing stuff?

File: src/NJsonSchema/Infrastructure/XmlObjectExtension.cs (hunk 1 above)
Comment (emilw) [reply src/NJsonSchema/Infrastructure/XmlObjectExtension.cs]: What i am after mainly is to detect if it is the first parameter to the controller or not, e.g. not a DTO property of type Array. I it is, the ArrayOf[Name] should be applied. Do you see any other way to detect that?

**Comment by emilw** [comment]
@RSuter , I have been running it end to end tests with the NJsonSchema incorparated with NSwag and executed our full pipleline and compared the full Swagger spec before and after and the only changes i found is that the ArrayOf* and the Int32 => int seems to work and no other changes.
There were some required parameter changes, but i think you have added that since our last NSwag upgrade....so i feel quite good :-).

**Comment by emilw** [comment]
Great, when do you think it's possible to get out a new version of NSwag with this change included?

File: src/NJsonSchema/Infrastructure/XmlObjectExtension.cs (hunk 1 above)
Comment (RSuter) [reply src/NJsonSchema/Infrastructure/XmlObjectExtension.cs]: So it's to detect if it's a direct usage in a parameter vs. part of a schema (e.g. property)? I think for this case, checking the ParentSchema is fine... let me know if you see problems in your app with this approach (I personally don't use xml schemas in any project).

**Comment by RSuter** [comment]
Releasing NJS v9.11.1

File: src/NJsonSchema/Infrastructure/XmlObjectExtension.cs (hunk 1 above)
Comment (emilw) [reply src/NJsonSchema/Infrastructure/XmlObjectExtension.cs]: Correct, thats the difference. Earlier I assumed that all arrays were properties on a DTO/Schema which is not correct....

**Pull request merged**